cmd/npmExecuteScripts.go

```diff
@@ -0,0 +1,204 @@
+	utils := npmExecuteScriptsUtilsBundle{}
+
+	err := runNpmExecuteScripts(&utils, &config)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+func runNpmExecuteScripts(utils npmExecuteScriptsUtilsInterface, options *npmExecuteScriptsOptions) error {
+	execRunner := utils.getExecRunner()
+	packageJSONFiles, err := findPackageJSONFiles(utils)
+	if err != nil {
+		return err
+	}
+
+	oldWorkingDirectory, err := utils.getwd()
+	if err != nil {
+		return err
+	}
+
+	for _, file := range packageJSONFiles {
+		dir := path.Dir(file)
+		err = utils.chdir(dir)
+		if err != nil {
+			return err
+		}
+
+		// set in each directory to respect existing config in rc files
+		err = setNpmRegistries(options, execRunner)
+		if err != nil {
+			return err
+		}
+
+		packageLockExists, yarnLockExists, err := checkIfLockFilesExist(utils)
+		if err != nil {
+			return err
+		}
+		if options.Install {
+			err = installDependencies(dir, packageLockExists, yarnLockExists, execRunner)
+			if err != nil {
+				return err
+			}
+		}
+
+		for _, v := range options.RunScripts {
+			log.Entry().WithField("WorkingDirectory", dir).Info("run-script " + v)
+			err = execRunner.RunExecutable("npm", "run-script", v, "--if-present")
+			if err != nil {
+				return err
+			}
+		}
+		err = utils.chdir(oldWorkingDirectory)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+func setNpmRegistries(options *npmExecuteScriptsOptions, execRunner execRunner) error {
+	environment := []string{}
+	const sapRegistry = "@sap:registry"
+	const npmRegistry = "registry"
+	configurableRegistries := []string{npmRegistry, sapRegistry}
+	for _, registry := range configurableRegistries {
+		var buffer bytes.Buffer
+		execRunner.Stdout(&buffer)
+		err := execRunner.RunExecutable("npm", "config", "get", registry)
+		execRunner.Stdout(log.Entry().Writer())
+		if err != nil {
+			return err
+		}
+		preConfiguredRegistry := buffer.String()
+
+		log.Entry().Info("Discovered pre-configured npm registry " + preConfiguredRegistry)
+
+		if registry == npmRegistry && options.DefaultNpmRegistry != "" && (preConfiguredRegistry == "undefined" || strings.HasPrefix(preConfiguredRegistry, "https://registry.npmjs.org")) {
+			log.Entry().Info("npm registry " + registry + " was not configured, setting it to " + options.DefaultNpmRegistry)
+			environment = append(environment, "npm_config_"+registry+"="+options.DefaultNpmRegistry)
+		}
+
+		if registry == sapRegistry && (preConfiguredRegistry == "undefined" || strings.HasPrefix(preConfiguredRegistry, "https://npm.sap.com")) {
+			log.Entry().Info("npm registry " + registry + " was not configured, setting it to " + options.SapNpmRegistry)
+			environment = append(environment, "npm_config_"+registry+"="+options.SapNpmRegistry)
+		}
+	}
+
+	if len(environment) > 0 {
+		log.Entry().Info("Setting environment: " + strings.Join(environment, ", "))
+		execRunner.SetEnv(environment)
+	}
```

Comment:
I am not 100% sure, but I think it was actually correct before to do it unconditionally. If you don't want to have any (additional) env, you should set an empty array. Now it could be the case that you are using the env from a previous loop-iteration in another folder.

Reply:
yes, have not considered the loop here. thanks.